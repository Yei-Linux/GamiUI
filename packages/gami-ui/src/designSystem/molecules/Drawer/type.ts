import { IGenericEvents } from 'core/domain/interfaces/IGeneralProps'
import { DrawerPlacementTypes } from 'core/domain/types'
import { TDesignUI } from 'core/domain/types/design-system'

export type TDrawer = {
  /**
   * Children Content
   */
  children?: React.ReactNode
  /**
   * Open or not of drawer
   */
  open?: boolean
  /**
   * Has Mask
   */
  withMask?: boolean
  /**
   * Custom Icon
   */
  customCloseIcon?: React.ReactNode | null
  /**
   * Has Close Icon
   */
  hasCloseIcon?: boolean
  /**
   * Height
   */
  height?: number
  /**
   * Width
   */
  width?: number
  /**
   * ZIndex
   */
  zIndex?: number
  /**
   * Placemente Drawer
   */
  placement?: DrawerPlacementTypes
  /**
   * Action on close drawer
   */
  onClose?: () => void
}

export type TDrawerComponent = TDrawer & TDesignUI & IGenericEvents
