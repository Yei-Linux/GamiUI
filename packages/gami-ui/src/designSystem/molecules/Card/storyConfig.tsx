import React from 'react'
import {
  IStoryConfig,
  IStoryMainConfig,
} from 'core/domain/interfaces/IStorybook'
import {
  getInheritGlobalStylesStories,
  getStoryConfigStructure,
  madegenericPropsControl,
} from 'core/helpers/storybook.helper'
import Card from '../Card'
import RichText from 'designSystem/atoms/RichText'
import Button from 'designSystem/atoms/Button'

const genericArgTypes = madegenericPropsControl([])

const storyArgTypes = {
  ...genericArgTypes,
}

const docArgTypes = {
  children: {
    control: 'object',
    description: 'Display Children content',
    table: {
      type: { summary: 'React.ReactNode' },
    },
  },
}

const argTypes = { ...storyArgTypes, ...docArgTypes }

const mainConfig: IStoryMainConfig = {
  title: 'Molecules/Card ðŸŸ¢',
  component: Card,
  args: {},
  argTypes: storyArgTypes,
}

const storiesInheritGlobalStyles = getInheritGlobalStylesStories({
  rounded: {
    args: {
      children: 'Gami Card',
      padding: '10px',
      shadow: 'xs',
      rounded: 'xs',
    },
  },
  shadow: {
    args: {
      children: 'Gami Card',
      padding: '10px',
      shadow: 'xs',
      rounded: 'xs',
    },
  },
  width: {
    args: {
      children: 'Gami Card',
      padding: '10px',
      shadow: 'xs',
      rounded: 'xs',
    },
  },
  height: {
    args: {
      children: 'Gami Card',
      padding: '10px',
      shadow: 'xs',
      rounded: 'xs',
    },
  },
  size: {
    args: {
      children: 'Gami Card',
      padding: '10px',
      shadow: 'xs',
      rounded: 'xs',
    },
  },
  fontWeight: {
    args: {
      children: 'Gami Card',
      padding: '10px',
      shadow: 'xs',
      rounded: 'xs',
    },
  },
  textAlign: {
    args: {
      width: 'lg',
      children: 'Gami Card',
      padding: '10px',
      shadow: 'xs',
      rounded: 'xs',
    },
  },
  margin: {
    args: {
      children: 'Gami Card',
      padding: '10px',
      shadow: 'xs',
      rounded: 'xs',
    },
    examples: ['0px', '2px', '4px', '6px', '8px', '10px'],
    wrapper: (content: React.ReactNode) => (
      <div style={{ border: '1px solid #d1d1d1' }}>{content}</div>
    ),
  },
  padding: {
    args: {
      children: 'Gami Card',
      shadow: 'xs',
      rounded: 'xs',
    },
    examples: ['0px', '2px', '4px', '6px', '8px', '10px'],
  },
})

const storiesComponent: IStoryConfig[] = [
  {
    storyName: 'With Basic ðŸ™‚',
    self: {
      args: {
        width: 'fit',
        shadow: 'xs',
        rounded: 'xs',
      },
      variants: {
        examples: [
          {
            label: 'Basic',
            value: <p style={{ padding: '1rem' }}>Hola como estas</p>,
          },
        ],
        field: 'children',
      },
    },
  },
  {
    storyName: 'With Content ðŸ™‚',
    self: {
      args: {
        width: 'fit',
        shadow: 'xs',
        rounded: 'xs',
      },
      variants: {
        examples: [
          {
            label: 'Content',
            value: (
              <Card.Content
                title={<RichText text="**Hi, this is my title**" />}
                description="Hi this is description"
              />
            ),
          },
        ],
        field: 'children',
      },
    },
  },
  {
    storyName: 'With Cover ðŸ™‚',
    self: {
      args: {
        width: 'fit',
        shadow: 'xs',
        rounded: 'xs',
      },
      variants: {
        examples: [
          {
            label: 'Cover',
            value: (
              <>
                <Card.Cover>
                  <img
                    width="100%"
                    src="https://gw.alipayobjects.com/zos/rmsportal/JiqGstEfoWAOHiTxclqi.png"
                  />
                </Card.Cover>
                <Card.Content
                  title="Hi this is title"
                  description="Hi this is description"
                />
              </>
            ),
          },
        ],
        field: 'children',
      },
    },
  },
  {
    storyName: 'With Footer ðŸ™‚',
    self: {
      args: {
        width: 'fit',
        shadow: 'xs',
        rounded: 'xs',
      },
      variants: {
        examples: [
          {
            label: 'Footer',
            value: (
              <>
                <Card.Cover>
                  <img
                    width="100%"
                    src="https://gw.alipayobjects.com/zos/rmsportal/JiqGstEfoWAOHiTxclqi.png"
                  />
                </Card.Cover>
                <Card.Content
                  title="Hi this is title"
                  description="Hi this is description"
                />
                <Card.Footer>
                  <Button
                    type="button"
                    rounded="sm"
                    height="auto"
                    variant="secondary"
                    width="auto"
                  >
                    Text
                  </Button>
                </Card.Footer>
              </>
            ),
          },
        ],
        field: 'children',
      },
    },
  },
]

const storyConfig = getStoryConfigStructure({
  mainConfig,
  storiesInheritGlobalStyles,
  storiesComponent,
  component: Card,
})

export { storyConfig, argTypes }
