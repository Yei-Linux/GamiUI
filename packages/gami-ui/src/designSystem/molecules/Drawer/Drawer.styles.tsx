import styled from '@emotion/styled'

export const CloseIconStyled = styled('div')(() => ({
  position: 'absolute',
  top: '8px',
  right: '10px',
}))

type TDrawerStyled = {
  $zIndex: number
  $width: number
  $height: number
}
export const DrawerStyled = styled('div')(
  ({ $height, $width, $zIndex }: TDrawerStyled) => ({
    background: '#fff',
    boxShadow: '0 -2px 8px rgb(0 0 0 / 15%)',
    overflowY: 'auto',
    position: 'fixed',
    zIndex: $zIndex,
    '&.left': {
      top: '0px',
      bottom: '0px',
      left: $width && `-${$width}px`,
      width: $width && `${$width}px`,
      borderTopRightRadius: '2em',
    },
    '&.right': {
      top: '0px',
      bottom: '0px',
      right: $width && `-${$width}px`,
      width: $width && `${$width}px`,
      borderTopLeftRadius: '2em',
    },
    '&.top': {
      top: $height && `-${$height}px`,
      bottom: 'auto',
      left: '0px',
      right: '0px',
      width: '100%',
      height: $height && `${$height}px`,
      borderBottomLeftRadius: '2em',
      borderBottomRightRadius: '2em',
    },
    '&.bottom': {
      top: 'auto',
      bottom: $height && `-${$height}px`,
      left: '0px',
      right: '0px',
      width: '100%',
      height: $height && `${$height}px`,
      borderTopLeftRadius: '2em',
      borderTopRightRadius: '2em',
    },
  })
)
