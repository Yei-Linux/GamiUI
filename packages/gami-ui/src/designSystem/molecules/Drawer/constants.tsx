export const drawerTranstionByStates = {
  mask: {
    open: {
      from: {
        duration: 1.5,
        opacity: 0.3,
      },
      to: {
        opacity: 1,
      },
    },
    close: {
      from: null,
      to: {
        duration: 0.5,
        opacity: 0.01,
      },
    },
  },
  drawer: {
    left: {
      open: {
        to: {
          duration: 0.8,
          delay: 0,
          position: { axis: 'xPercent', value: 100 },
        },
      },
      close: {
        to: {
          duration: 0.8,
          delay: 0,
          position: { axis: 'xPercent', value: -100 },
        },
      },
    },
    right: {
      open: {
        to: {
          duration: 0.8,
          delay: 0,
          position: { axis: 'xPercent', value: -100 },
        },
      },
      close: {
        to: {
          duration: 0.8,
          delay: 0,
          position: { axis: 'xPercent', value: 100 },
        },
      },
    },
    top: {
      open: {
        to: {
          duration: 0.8,
          delay: 0,
          position: { axis: 'yPercent', value: 100 },
        },
      },
      close: {
        to: {
          duration: 0.8,
          delay: 0,
          position: { axis: 'yPercent', value: -100 },
        },
      },
    },
    bottom: {
      open: {
        to: {
          duration: 0.8,
          delay: 0,
          position: { axis: 'yPercent', value: -100 },
        },
      },
      close: {
        to: {
          duration: 0.8,
          delay: 0,
          position: { axis: 'yPercent', value: 100 },
        },
      },
    },
  },
}

export const options = {
  placementType: ['left', 'right', 'top', 'bottom'] as const,
  type: [] as const,
}
