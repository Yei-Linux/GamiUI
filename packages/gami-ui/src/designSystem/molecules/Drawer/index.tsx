export { default } from './Drawer'
export { options } from './constants'
export { argTypes } from './storyConfig'
