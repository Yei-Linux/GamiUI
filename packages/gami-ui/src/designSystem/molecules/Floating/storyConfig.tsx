import {
  getInheritGlobalStylesStories,
  getStoryConfigStructure,
} from 'core/helpers/storybook.helper'
import Floating from '.'
import {
  IStoryConfig,
  IStoryMainConfig,
} from 'core/domain/interfaces/IStorybook'

const mainConfig: IStoryMainConfig = {
  title: 'Molecules/Floating',
  component: Floating,
  args: {},
  argTypes: {},
  parameters: {
    actions: { argTypesRegex: '^on[A-Z].*' },
    docs: {
      source: {
        type: 'code',
      },
    },
  },
}

const storiesInheritGlobalStyles = getInheritGlobalStylesStories({
  rounded: null,
  shadow: null,
  width: null,
  height: null,
  size: null,
  fontWeight: null,
  textAlign: null,
  margin: null,
  padding: null,
})

const storiesComponent: IStoryConfig[] = []

const storyConfig = getStoryConfigStructure({
  mainConfig,
  storiesInheritGlobalStyles,
  storiesComponent,
  component: Floating,
})

export { storyConfig }
