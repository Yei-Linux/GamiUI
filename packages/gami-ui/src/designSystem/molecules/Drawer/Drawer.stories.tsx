import React, { useState } from 'react'
import { Meta, ComponentStory } from '@storybook/react'
import Button from 'designSystem/atoms/Button'
import { TJSXElements } from 'core/domain/interfaces/common'

import { storyConfig } from './storyConfig'
import Drawer from '.'
import { DrawerPlacementTypes } from 'core/domain/types'
import Row from 'designSystem/layouts/Row'
import Icon from 'designSystem/atoms/Icon'
import RichText from 'designSystem/atoms/RichText'

const { mainConfig, stories } = storyConfig

export default mainConfig as Meta

// eslint-disable-next-line no-empty-pattern
const [] = stories

const StoryDefault = (args: any) => {
  const [open, setOpen] = useState(false)

  const toggle = () => setOpen(!open)

  return (
    <Row gap="10px">
      <Button type="button" width="fit" onClick={toggle}>
        Toggle
      </Button>
      <Drawer onClose={() => setOpen(!open)} open={open} {...args}>
        <RichText text="Hey, whats up!" />
      </Drawer>
    </Row>
  )
}
export const StoryDefaultTemplate: ComponentStory<TJSXElements> =
  StoryDefault.bind({})
StoryDefaultTemplate.storyName = 'WithDefault'
StoryDefaultTemplate.args = {
  height: 300,
  width: 280,
  zIndex: 2,
}

const StoryCustomCloseIcon = (args: any) => {
  const [customCloseIcon, setCustomCloseIcon] = useState(true)
  const [open, setOpen] = useState(false)

  const toggle = (hasMask: boolean) => {
    setCustomCloseIcon(hasMask)
    setOpen(!open)
  }

  return (
    <Row gap="10px">
      <Button type="button" width="fit" onClick={() => toggle(true)}>
        CustomCloseIcon
      </Button>
      <Button type="button" width="fit" onClick={() => toggle(false)}>
        Not CustomCloseIcon
      </Button>
      <Drawer
        customCloseIcon={
          customCloseIcon ? (
            <Icon name="close" color="black" size="20px" />
          ) : null
        }
        onClose={() => setOpen(!open)}
        open={open}
        {...args}
      >
        <RichText text="Hey, whats up!" />
      </Drawer>
    </Row>
  )
}
export const StoryCustomCloseIconTemplate: ComponentStory<TJSXElements> =
  StoryCustomCloseIcon.bind({})

StoryCustomCloseIconTemplate.storyName = 'WithCustomCloseIcon'
StoryCustomCloseIconTemplate.args = {
  hasCloseIcon: true,
  placement: 'left',
  height: 300,
  width: 280,
  zIndex: 2,
}

const StoryCloseIcon = (args: any) => {
  const [closeIcon, setCloseIcon] = useState(true)
  const [open, setOpen] = useState(false)

  const toggle = (hasMask: boolean) => {
    setCloseIcon(hasMask)
    setOpen(!open)
  }

  return (
    <Row gap="10px">
      <Button type="button" width="fit" onClick={() => toggle(true)}>
        Close Icon
      </Button>
      <Button type="button" width="fit" onClick={() => toggle(false)}>
        Not Close Icon
      </Button>
      <Drawer
        hasCloseIcon={closeIcon}
        onClose={() => setOpen(!open)}
        open={open}
        {...args}
      >
        <RichText text="Hey, whats up!" />
      </Drawer>
    </Row>
  )
}
export const StoryCloseIconTemplate: ComponentStory<TJSXElements> =
  StoryCloseIcon.bind({})

StoryCloseIconTemplate.storyName = 'WithCloseIcon'
StoryCloseIconTemplate.args = {
  placement: 'left',
  height: 300,
  width: 280,
  zIndex: 2,
}

const StoryMask = (args: any) => {
  const [mask, setMask] = useState(true)
  const [open, setOpen] = useState(false)

  const toggle = (hasMask: boolean) => {
    setMask(hasMask)
    setOpen(!open)
  }

  return (
    <Row gap="10px">
      <Button type="button" width="fit" onClick={() => toggle(true)}>
        Mask
      </Button>
      <Button type="button" width="fit" onClick={() => toggle(false)}>
        Not Mask
      </Button>
      <Drawer
        withMask={mask}
        onClose={() => setOpen(!open)}
        open={open}
        {...args}
      >
        <RichText text="Hey, whats up!" />
      </Drawer>
    </Row>
  )
}
export const StoryMaskTemplate: ComponentStory<TJSXElements> = StoryMask.bind(
  {}
)

StoryMaskTemplate.storyName = 'WithMask'
StoryMaskTemplate.args = {
  height: 300,
  width: 280,
  zIndex: 2,
}

const StoryDirection = (args: any) => {
  const [placement, setPlacement] = useState<DrawerPlacementTypes>('left')
  const [open, setOpen] = useState(false)

  const toggle = (placementProp: DrawerPlacementTypes) => {
    setPlacement(placementProp)
    setOpen(!open)
  }

  return (
    <Row gap="10px">
      <Button type="button" width="fit" onClick={() => toggle('left')}>
        Left
      </Button>
      <Button type="button" width="fit" onClick={() => toggle('right')}>
        Right
      </Button>
      <Button type="button" width="fit" onClick={() => toggle('top')}>
        Top
      </Button>
      <Button type="button" width="fit" onClick={() => toggle('bottom')}>
        Bottom
      </Button>
      <Drawer
        onClose={() => setOpen(!open)}
        open={open}
        placement={placement}
        {...args}
      >
        <RichText text="Hey, whats up!" />
      </Drawer>
    </Row>
  )
}
export const StoryDirectionTemplate: ComponentStory<TJSXElements> =
  StoryDirection.bind({})
StoryDirectionTemplate.storyName = 'WithDirection'
StoryDirectionTemplate.args = {
  height: 300,
  width: 280,
  zIndex: 2,
}
