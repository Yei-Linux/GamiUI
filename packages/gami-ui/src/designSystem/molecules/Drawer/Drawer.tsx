import 'regenerator-runtime/runtime'

import React, { Fragment } from 'react'

import { drawerTranstionByStates } from './constants'
import * as S from './Drawer.styles'
import useOpen from 'hooks/useOpen'

import Mask from 'designSystem/atoms/Mask'
import Transition from 'designSystem/styled/Transition'

export interface IDrawer {
  /**
   * Custom Styles for drawer
   */
  style?: React.CSSProperties
  /**
   * Children Content
   */
  children?: React.ReactNode
  /**
   * Open or not of drawer
   */
  open?: boolean
  /**
   * Placement of drawer
   */
  placement?: string
  /**
   * Width
   */
  width?: string
  /**
   * Height
   */
  height?: string
  /**
   * Has Mask
   */
  withMask?: boolean
  /**
   * Action on close drawer
   */
  onClose?: () => void
}

const Drawer = ({
  children,
  open,
  onClose,
  withMask = true,
  style = {},
}: IDrawer) => {
  const { isOpen } = useOpen({ open })

  return (
    <Fragment>
      {isOpen && (
        <Fragment>
          <Transition
            from={drawerTranstionByStates.mask[open ? 'open' : 'close']?.from}
            to={drawerTranstionByStates.mask[open ? 'open' : 'close'].to}
            isReadyToInitAnimation={open}
          >
            <div>
              <Mask
                onClick={onClose}
                background={withMask ? 'rgba(0, 0, 0, 0.45)' : 'transparent'}
              />
            </div>
          </Transition>

          <Transition
            to={drawerTranstionByStates.drawer[open ? 'open' : 'close'].to}
            isReadyToInitAnimation={open}
          >
            <S.Drawer style={style}>{children}</S.Drawer>
          </Transition>
        </Fragment>
      )}
    </Fragment>
  )
}

export default Drawer
