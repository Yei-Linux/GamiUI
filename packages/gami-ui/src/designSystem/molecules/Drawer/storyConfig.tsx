import {
  getInheritGlobalStylesStories,
  getStoryConfigStructure,
  madegenericPropsControl,
} from 'core/helpers/storybook.helper'
import Drawer, { options } from '.'
import {
  IStoryConfig,
  IStoryMainConfig,
} from 'core/domain/interfaces/IStorybook'

const genericArgTypes = madegenericPropsControl([
  'size',
  'fontWeight',
  'textAlign',
  'margin',
  'padding',
  'rounded',
  'width',
  'height',
])

const storyArgTypes = {
  open: {
    control: 'boolean',
    description: 'Open or not of drawer',
    table: {
      type: { summary: 'boolean' },
    },
  },
  withMask: {
    control: 'boolean',
    description: 'Has Mask',
    table: {
      type: { summary: 'boolean' },
    },
  },
  hasCloseIcon: {
    control: 'boolean',
    description: 'Has Close Icon',
    table: {
      type: { summary: 'boolean' },
    },
  },
  height: {
    control: 'number',
    description: 'Height',
    table: {
      type: { summary: 'number' },
    },
  },
  width: {
    control: 'number',
    description: 'Width',
    table: {
      type: { summary: 'number' },
    },
  },
  zIndex: {
    control: 'number',
    description: 'ZIndex',
    table: {
      type: { summary: 'number' },
    },
  },
  placement: {
    control: { type: 'select', options: options.placementType },
    description: 'Placemente Drawer',
    table: {
      type: { summary: 'string' },
    },
  },
  ...genericArgTypes,
}

const docArgTypes = {
  children: {
    control: 'object',
    description: 'Children Prop',
    table: {
      type: { summary: 'React.ReactNode' },
    },
  },
  customCloseIcon: {
    control: 'object',
    description: 'Custom Icon',
    table: {
      type: { summary: 'React.ReactNode' },
    },
  },
  onClose: {
    control: 'object',
    description: 'CAction on close drawer',
    table: {
      type: { summary: '() => void' },
    },
  },
}

const argTypes = { ...storyArgTypes, ...docArgTypes }

const mainConfig: IStoryMainConfig = {
  title: 'Molecules/Drawer',
  component: Drawer,
  args: {},
  argTypes: storyArgTypes,
  parameters: {
    actions: { argTypesRegex: '^on[A-Z].*' },
    docs: {
      source: {
        type: 'code',
      },
    },
  },
}

const storiesInheritGlobalStyles = getInheritGlobalStylesStories({
  rounded: null,
  shadow: null,
  width: null,
  height: null,
  size: null,
  fontWeight: null,
  textAlign: null,
  margin: null,
  padding: null,
})

const storiesComponent: IStoryConfig[] = []

const storyConfig = getStoryConfigStructure({
  mainConfig,
  storiesInheritGlobalStyles,
  storiesComponent,
  component: Drawer,
})

export { storyConfig, argTypes }
