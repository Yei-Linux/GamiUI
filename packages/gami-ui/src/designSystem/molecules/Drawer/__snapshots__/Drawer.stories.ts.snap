// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/Drawer Basic 1`] = `
<div>
  <button
    className="css-18i0sof-ButtonWrapper ekfoxpm0"
    onClick={[Function]}
    type="submit"
    width="SMALL"
  >
    Open Drawer
  </button>
  <div>
    <div
      className="css-1528g5q-MaskWrapper e1qqpb1u0"
      onClick={[Function]}
    />
  </div>
  <div
    className="css-ursirt-DrawerWrapper evpjr7q0"
  >
    <p>
      Hey, whats up
    </p>
  </div>
</div>
`;
